[BUG] LDAP auth broke after upgrading from opensearch:2 to opensearch:3

**What is the bug?**
There are two methods to integrate LDAP auth documented here: https://docs.opensearch.org/latest/security/authentication-backends/ldap/#approach-1-query-the-role-subtree
https://docs.opensearch.org/latest/security/authentication-backends/ldap/#approach-2-use-a-users-attribute-as-the-role-name

Approach 2 no longer works after upgrading. 
The LDAP server(FreeIPA) has all group memberships under the user as memberOf attribute. 

**How can one reproduce the bug?**
Steps to reproduce the behavior:
Use a config similar to: 
```
      authz:
        ldap_roles:
          description: "Authorize using LDAP"
          http_enabled: true
          transport_enabled: true
          authorization_backend:
            type: ldap
            config:
              enable_ssl: false
              enable_start_tls: false
              enable_ssl_client_auth: false
              verify_hostnames: true
              hosts:
                - "freeipa-1:389"
              bind_dn: uid=opensearch,cn=users,cn=accounts,dc=example,dc=com
              password: "${password}"
              userbase: cn=users,cn=accounts,dc=example,dc=com
              usersearch: (uid={0})
              username_attribute: uid
              skip_users:
                - admin
                - kibanaserver
              userroleattribute: memberOf
              rolesearch_enabled: false
              rolebase: cn=groups,cn=accounts,dc=example,dc=com
              rolename: cn
```

**What is the expected behavior?**
Roles to be resolved as they were in opensearch:2

**What is your host/environment?**
 - OS: Linux (docker)
 - Version: 9
 - Plugins (default in the image)

**Do you have any additional context?**
Related Forum posts: 
https://forum.opensearch.org/t/ldap-login-does-not-work-after-upgrade-from-2-19-1-to-3-1-0/25603
https://forum.opensearch.org/t/solved-unable-to-log-in-with-ldap-after-upgrade-to-3-3/27360/4

@nibix I have tried using userroleattribute: memberOf on the 3.5.0 and still get NPE. 
```
{"type": "opensearch-log", "timestamp": "2026-02-12T17:50:11,978+0000", "level": "TRACE", "component": "o.o.s.a.l.LDAPAuthorizationBackend2", "cluster.name": "test.opensearch", "node.name": "opensearch-test", "message": "roles count total 64", "cluster.uuid": "il4Co70jQUCrUzCJn-bDiw", "node.id": "HIBvUttySYSEkdFs6Kpgww"  }
{"type": "opensearch-log", "timestamp": "2026-02-12T17:50:12,457+0000", "level": "DEBUG", "component": "o.o.s.a.l.LDAPAuthorizationBackend2", "cluster.name": "test.opensearch", "node.name": "opensearch-test", "message": "Roles for dforster -\u003e []", "cluster.uuid": "il4Co70jQUCrUzCJn-bDiw", "node.id": "HIBvUttySYSEkdFs6Kpgww"  }
{"type": "opensearch-log", "timestamp": "2026-02-12T17:50:12,457+0000", "level": "TRACE", "component": "o.o.s.a.l.LDAPAuthorizationBackend2", "cluster.name": "test.opensearch", "node.name": "opensearch-test", "message": "returned user: User [name=dforster, backend_roles=[], requestedTenant=null]", "cluster.uuid": "il4Co70jQUCrUzCJn-bDiw", "node.id": "HIBvUttySYSEkdFs6Kpgww"  }
{"type": "opensearch-log", "timestamp": "2026-02-12T17:50:12,457+0000", "level": "DEBUG", "component": "o.o.s.a.l.LDAPAuthorizationBackend2", "cluster.name": "test.opensearch", "node.name": "opensearch-test", "message": "Unable to fill user roles due to ", "cluster.uuid": "il4Co70jQUCrUzCJn-bDiw", "node.id": "HIBvUttySYSEkdFs6Kpgww", 
"stacktrace": ["java.lang.NullPointerException",
"at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:906) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.collect.ImmutableSet$Builder.add(ImmutableSet.java:520) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.collect.ImmutableSet$Builder.add(ImmutableSet.java:473) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.collect.ImmutableCollection$Builder.addAll(ImmutableCollection.java:477) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.collect.ImmutableSet$Builder.addAll(ImmutableSet.java:544) ~[guava-33.2.1-jre.jar:?]",
"at org.opensearch.security.user.User.withRoles(User.java:175) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at org.opensearch.security.auth.ldap2.LDAPAuthorizationBackend2.addRoles0(LDAPAuthorizationBackend2.java:390) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at org.opensearch.security.auth.ldap2.LDAPAuthorizationBackend2.lambda$addRoles$0(LDAPAuthorizationBackend2.java:123) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at org.opensearch.secure_sm.AccessController.doPrivilegedChecked(AccessController.java:110) ~[opensearch-agent-policy-3.5.0.jar:3.5.0]",
"at org.opensearch.security.auth.ldap2.LDAPAuthorizationBackend2.addRoles(LDAPAuthorizationBackend2.java:123) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at org.opensearch.security.auth.BackendRegistry.authz(BackendRegistry.java:746) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at org.opensearch.security.auth.BackendRegistry$5.call(BackendRegistry.java:797) ~[?:?]",
"at org.opensearch.security.auth.BackendRegistry$5.call(BackendRegistry.java:786) ~[?:?]",
"at com.google.common.cache.LocalCache$LocalManualCache$1.load(LocalCache.java:4903) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3574) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2316) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2189) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2079) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.cache.LocalCache.get(LocalCache.java:4017) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4898) ~[guava-33.2.1-jre.jar:?]",
"at org.opensearch.security.auth.BackendRegistry.authcz(BackendRegistry.java:786) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at org.opensearch.security.auth.BackendRegistry.authenticate(BackendRegistry.java:424) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at org.opensearch.security.filter.SecurityRestFilter.checkAndAuthenticateRequest(SecurityRestFilter.java:365) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at org.opensearch.security.ssl.http.netty.Netty4HttpRequestHeaderVerifier.channelRead0(Netty4HttpRequestHeaderVerifier.java:90) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at org.opensearch.security.ssl.http.netty.Netty4HttpRequestHeaderVerifier.channelRead0(Netty4HttpRequestHeaderVerifier.java:37) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:356) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:361) ~[netty-codec-base-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:325) ~[netty-codec-base-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:356) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:293) ~[netty-handler-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:354) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:107) ~[netty-codec-base-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:356) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1540) ~[netty-handler-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1398) ~[netty-handler-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1449) ~[netty-handler-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:545) ~[netty-codec-base-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:484) ~[netty-codec-base-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296) ~[netty-codec-base-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:356) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1429) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:918) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:172) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.handle(AbstractNioChannel.java:445) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.nio.NioIoHandler$DefaultNioRegistration.handle(NioIoHandler.java:388) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.nio.NioIoHandler.processSelectedKey(NioIoHandler.java:596) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.nio.NioIoHandler.processSelectedKeysPlain(NioIoHandler.java:541) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.nio.NioIoHandler.processSelectedKeys(NioIoHandler.java:514) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.nio.NioIoHandler.run(NioIoHandler.java:484) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.SingleThreadIoEventLoop.runIo(SingleThreadIoEventLoop.java:225) [netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.SingleThreadIoEventLoop.run(SingleThreadIoEventLoop.java:196) [netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:1195) [netty-common-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.2.9.Final.jar:4.2.9.Final]",
"at java.base/java.lang.Thread.run(Thread.java:1474) [?:?]"] }
{"type": "opensearch-log", "timestamp": "2026-02-12T17:50:12,459+0000", "level": "ERROR", "component": "o.o.s.a.BackendRegistry", "cluster.name": "test.opensearch", "node.name": "opensearch-test", "message": "Cannot retrieve roles for User [name=dforster, backend_roles=[], requestedTenant=null] from ldap due to OpenSearchSecurityException[java.lang.NullPointerException]; nested: NullPointerException;", "cluster.uuid": "il4Co70jQUCrUzCJn-bDiw", "node.id": "HIBvUttySYSEkdFs6Kpgww" , 
"stacktrace": ["org.opensearch.OpenSearchSecurityException: java.lang.NullPointerException",
"at org.opensearch.security.auth.ldap2.LDAPAuthorizationBackend2.addRoles0(LDAPAuthorizationBackend2.java:395) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at org.opensearch.security.auth.ldap2.LDAPAuthorizationBackend2.lambda$addRoles$0(LDAPAuthorizationBackend2.java:123) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at org.opensearch.secure_sm.AccessController.doPrivilegedChecked(AccessController.java:110) ~[opensearch-agent-policy-3.5.0.jar:3.5.0]",
"at org.opensearch.security.auth.ldap2.LDAPAuthorizationBackend2.addRoles(LDAPAuthorizationBackend2.java:123) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at org.opensearch.security.auth.BackendRegistry.authz(BackendRegistry.java:746) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at org.opensearch.security.auth.BackendRegistry$5.call(BackendRegistry.java:797) ~[?:?]",
"at org.opensearch.security.auth.BackendRegistry$5.call(BackendRegistry.java:786) ~[?:?]",
"at com.google.common.cache.LocalCache$LocalManualCache$1.load(LocalCache.java:4903) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3574) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2316) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2189) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2079) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.cache.LocalCache.get(LocalCache.java:4017) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.cache.LocalCache$LocalManualCache.get(LocalCache.java:4898) ~[guava-33.2.1-jre.jar:?]",
"at org.opensearch.security.auth.BackendRegistry.authcz(BackendRegistry.java:786) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at org.opensearch.security.auth.BackendRegistry.authenticate(BackendRegistry.java:424) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at org.opensearch.security.filter.SecurityRestFilter.checkAndAuthenticateRequest(SecurityRestFilter.java:365) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at org.opensearch.security.ssl.http.netty.Netty4HttpRequestHeaderVerifier.channelRead0(Netty4HttpRequestHeaderVerifier.java:90) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at org.opensearch.security.ssl.http.netty.Netty4HttpRequestHeaderVerifier.channelRead0(Netty4HttpRequestHeaderVerifier.java:37) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:356) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:361) ~[netty-codec-base-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:325) ~[netty-codec-base-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:356) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:293) ~[netty-handler-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:354) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:107) ~[netty-codec-base-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:356) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1540) ~[netty-handler-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1398) ~[netty-handler-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1449) ~[netty-handler-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:545) ~[netty-codec-base-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:484) ~[netty-codec-base-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296) ~[netty-codec-base-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:356) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1429) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:918) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:172) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.handle(AbstractNioChannel.java:445) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.nio.NioIoHandler$DefaultNioRegistration.handle(NioIoHandler.java:388) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.nio.NioIoHandler.processSelectedKey(NioIoHandler.java:596) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.nio.NioIoHandler.processSelectedKeysPlain(NioIoHandler.java:541) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.nio.NioIoHandler.processSelectedKeys(NioIoHandler.java:514) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.nio.NioIoHandler.run(NioIoHandler.java:484) ~[netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.SingleThreadIoEventLoop.runIo(SingleThreadIoEventLoop.java:225) [netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.channel.SingleThreadIoEventLoop.run(SingleThreadIoEventLoop.java:196) [netty-transport-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:1195) [netty-common-4.2.9.Final.jar:4.2.9.Final]",
"at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.2.9.Final.jar:4.2.9.Final]",
"at java.base/java.lang.Thread.run(Thread.java:1474) [?:?]",
"Caused by: java.lang.NullPointerException",
"at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:906) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.collect.ImmutableSet$Builder.add(ImmutableSet.java:520) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.collect.ImmutableSet$Builder.add(ImmutableSet.java:473) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.collect.ImmutableCollection$Builder.addAll(ImmutableCollection.java:477) ~[guava-33.2.1-jre.jar:?]",
"at com.google.common.collect.ImmutableSet$Builder.addAll(ImmutableSet.java:544) ~[guava-33.2.1-jre.jar:?]",
"at org.opensearch.security.user.User.withRoles(User.java:175) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"at org.opensearch.security.auth.ldap2.LDAPAuthorizationBackend2.addRoles0(LDAPAuthorizationBackend2.java:390) ~[opensearch-security-3.5.0.0.jar:3.5.0.0]",
"... 48 more"] }
{"type": "opensearch-log", "timestamp": "2026-02-12T17:50:12,461+0000", "level": "DEBUG", "component": "o.o.s.a.BackendRegistry", "cluster.name": "test.opensearch", "node.name": "opensearch-test", "message": "Rest user 'User [name=dforster, backend_roles=[], requestedTenant=null]' is authenticated", "cluster.uuid": "il4Co70jQUCrUzCJn-bDiw", "node.id": "HIBvUttySYSEkdFs6Kpgww"  }
```
Is there a better way for me to test updates to ensure the PR actually fixes the issue? I run opensearch in docker.

--- a/src/main/java/org/opensearch/security/auth/ldap/backend/LDAPAuthorizationBackend.java
+++ b/src/main/java/org/opensearch/security/auth/ldap/backend/LDAPAuthorizationBackend.java
@@ -936,6 +936,10 @@ public class LDAPAuthorizationBackend implements AuthorizationBackend {
 
             } else {
                 // DN roles, extract rolename according to config
+                if (connection == null) {
+                    connection = getConnection(settings, configPath);
+                }
+
                 for (final LdapName roleLdapName : ldapRoles) {
                     final String role = getRoleFromEntry(connection, roleLdapName, roleName);
 

--- a/src/main/java/org/opensearch/security/auth/ldap/backend/LDAPAuthorizationBackend.java
+++ b/src/main/java/org/opensearch/security/auth/ldap/backend/LDAPAuthorizationBackend.java
@@ -832,15 +832,15 @@ public class LDAPAuthorizationBackend implements AuthorizationBackend {
                 userRoleAttributeValue = Utils.getSingleStringValue(userRoleAttribute);
             }
 
+            if (connection == null) {
+                connection = getConnection(settings, configPath);
+            }
+
             if (rolesearchEnabled) {
                 String escapedDn = dn;
 
                 if (isDebugEnabled) {
                     log.debug("DBGTRACE (8): escapedDn" + escapedDn);
-                }
-
-                if (connection == null) {
-                    connection = getConnection(settings, configPath);
                 }
 
                 for (Map.Entry<String, Settings> roleSearchSettingsEntry : roleBaseSettings) {
@@ -883,10 +883,6 @@ public class LDAPAuthorizationBackend implements AuthorizationBackend {
 
             if (isTraceEnabled) {
                 log.trace("roles count total {}", ldapRoles.size());
-            }
-
-            if (connection == null) {
-                connection = getConnection(settings, configPath);
             }
 
             // nested roles, makes only sense for DN style role names

--- a/src/main/java/org/opensearch/security/auth/ldap/backend/LDAPAuthorizationBackend.java
+++ b/src/main/java/org/opensearch/security/auth/ldap/backend/LDAPAuthorizationBackend.java
@@ -885,6 +885,10 @@ public class LDAPAuthorizationBackend implements AuthorizationBackend {
                 log.trace("roles count total {}", ldapRoles.size());
             }
 
+            if (connection == null) {
+                connection = getConnection(settings, configPath);
+            }
+
             // nested roles, makes only sense for DN style role names
             if (nestedRoleMatcher != null) {
 
@@ -900,10 +904,6 @@ public class LDAPAuthorizationBackend implements AuthorizationBackend {
                     if (nameRoleSearchBaseKeys == null) {
                         log.error("Could not find roleSearchBaseKeys for " + roleLdapName + "; existing: " + resultRoleSearchBaseKeys);
                         continue;
-                    }
-
-                    if (connection == null) {
-                        connection = getConnection(settings, configPath);
                     }
 
                     final Set<LdapName> nestedRoles = resolveNestedRoles(
@@ -936,10 +936,6 @@ public class LDAPAuthorizationBackend implements AuthorizationBackend {
 
             } else {
                 // DN roles, extract rolename according to config
-                if (connection == null) {
-                    connection = getConnection(settings, configPath);
-                }
-
                 for (final LdapName roleLdapName : ldapRoles) {
                     final String role = getRoleFromEntry(connection, roleLdapName, roleName);
 

--- a/src/test/java/org/opensearch/security/auth/ldap/LdapBackendTestNewStyleConfig.java
+++ b/src/test/java/org/opensearch/security/auth/ldap/LdapBackendTestNewStyleConfig.java
@@ -726,6 +726,29 @@ public class LdapBackendTestNewStyleConfig {
     }
 
     @Test
+    public void testLdapAuthorizationRolesearchDisabledWithLdapAuthContext() throws Exception {
+        final Settings settings = Settings.builder()
+            .putList(ConfigConstants.LDAP_HOSTS, "localhost:" + ldapPort)
+            .put("users.u1.search", "(uid={0})")
+            .put("users.u1.base", "ou=people,o=TEST")
+            .put("roles.g1.base", "ou=groups,o=TEST")
+            .put(ConfigConstants.LDAP_AUTHZ_ROLENAME, "cn")
+            .put(ConfigConstants.LDAP_AUTHZ_ROLESEARCH_ENABLED, false)
+            .put(ConfigConstants.LDAP_AUTHZ_USERROLENAME, "description")
+            .build();
+
+        AuthenticationContext context = ctx("spock", "spocksecret");
+        User user = new LDAPAuthenticationBackend(settings, null).authenticate(context);
+        user = new LDAPAuthorizationBackend(settings, null).addRoles(user, context);
+
+        Assert.assertNotNull(user);
+        assertThat(user.getName(), is("cn=Captain Spock,ou=people,o=TEST"));
+        assertThat(user.getRoles().size(), is(2));
+        Assert.assertTrue(user.getRoles().contains("dummyempty"));
+        Assert.assertTrue(user.getRoles().contains("rolemo4"));
+    }
+
+    @Test
     public void testCustomAttributes() throws Exception {
 
         Settings settings = Settings.builder()

